CMakeLists.txt

```diff
@@ -156,20 +157,49 @@ set_source_files_properties("thirdparty/loguru/loguru.cpp"
 find_package(Threads REQUIRED)
 
 # create the lslboost target
-add_library(lslboost OBJECT
-	lslboost/serialization_objects.cpp
-)
-target_link_libraries(lslboost PUBLIC Threads::Threads)
-target_compile_features(lslboost PUBLIC cxx_std_11 cxx_lambda_init_captures)
-
-target_compile_definitions(lslboost
-	PUBLIC
-		BOOST_ALL_NO_LIB
-		BOOST_ASIO_STANDALONE
-		BOOST_THREAD_DONT_PROVIDE_INTERRUPTIONS
-)
-target_include_directories(lslboost SYSTEM PUBLIC
-	${CMAKE_CURRENT_SOURCE_DIR}/lslboost)
+if(LSL_BUNDLED_BOOST)
+	message(STATUS "Using bundled Boost")
+		
+	add_library(lslboost OBJECT
+		lslboost/serialization_objects.cpp
+	)
+	target_link_libraries(lslboost PUBLIC Threads::Threads)
+	target_compile_features(lslboost PUBLIC cxx_std_11 cxx_lambda_init_captures)
+
+	target_compile_definitions(lslboost
```

Comment:
This block should be after the `if(LSL_BUNDLED_BOOST)` as it is mostly identical to the one in the branch below (`PUBLIC` sets flags for the targets `PRIVATE` compilation as well as the `INTERFACE` flags, so it doesn't hurt to reuse this for a target without any source files of its own).

Reply:
I moved the block further down as part of b9d2b96b57feda4f244f2b6002e8225497a4b337. Please have a look if this is what you proposed.

Reply:
This fails with these errors:
>CMake Error at CMakeLists.txt:191 (target_compile_definitions):
  target_compile_definitions may only set INTERFACE properties on INTERFACE
  targets


>CMake Error at CMakeLists.txt:224 (target_link_libraries):
  INTERFACE library can only be used with the INTERFACE keyword of
  target_link_libraries


>CMake Error at CMakeLists.txt:229 (target_compile_definitions):
  target_compile_definitions may only set INTERFACE properties on INTERFACE
  targets

Reply:
I reverted the changes in 512109086acc608acd1851f69d971a681812a8e4 and also fixed some other related issues. This version works for me when using separately installed boost libraries.

Reply:
Ah, too bad. I think there's a more elegant solution for that, but I won't get around to try it this week.
I like the direction this is taking, but I need some more time to see if there's anything left to make it clearer / shorter.

Reply:
Thanks, take your time. I guess it would be more concise to do it how it was previously, without splitting up the OS-dependent flags.